org.OpenT2T.Sample.SuperPopular.Thermostat/Wink Thermostat/js/thingTranslator.js

```diff
@@ -0,0 +1,138 @@
+'use strict';
+
+var https = require('https');
+var wh = require("opent2t-translator-helper-wink"); //link locally for now
+
+// logs device state
+function logDeviceState(device) {
+    if (typeof (device) !== 'undefined') {
+        console.log('  device.name          : ' + device.name);
+        console.log('  device.props         : ' + device.props);
+    } else {
+        console.log('device is undefined');
+    }
+};
+
+var deviceId, accessToken;
+
+// module exports, implementing the schema
+module.exports = {
+
+    device: null,
+
+    initDevice: function(dev) {
+        this.device = dev;
+
+        if (typeof this.device != 'undefined') {
+            if (typeof (this.device.props) !== 'undefined') {
+                var props = JSON.parse(this.device.props);
+
+                if (typeof (props.access_token) !== 'undefined') {
+                    accessToken = props.access_token;
+                } else {
+                    console.log('props.access_token is undefined.');
+                }
+
+                if (typeof (props.id) !== 'undefined') {
+                    deviceId = props.id;
+                } else {
+                    console.log('props.id is undefined.');
+                }
+            } else {
+                console.log('props is undefined.');
+            }
+        } else {
+            console.log('device is undefined.');
+        }
+        //use the winkHub device and initialize
+        wh.initWinkApi("thermostats",deviceId,accessToken);
+        console.log('Javascript and Wink Helper initialized.');
+        logDeviceState(this.device);
+    },
+
+    turnOn: function() {
+        console.log('turnOn called.');
+
+        wh.sendDesiredStateCommand('powered',true).then(result => {
+        console.log(result);}).catch(error => {console.log(error.message);}); 
```

Comment:
This needs to return the promise, and also after logging return the result / re-throw the error. Since this pattern is repeated many times, how about a helper function:
````
function logPromise(p) {
    return p.then(result => {
        console.log(result);
        return result;
    }).catch(error => {
        console.log(error.message);
        throw error;
    });
}
```

Then use it like this:
```
return logPromise(wh.sendDesiredStateCommand('powered', true));